sotodlib/preprocess/core.py

```diff
@@ -146,6 +148,122 @@ def register(process_class):
                 f"Preprocess Module of name {name} is already Registered"
             )
 
+def _zeros_cls( item ):
+    """return a callable zeros class that exactly matches type item for use 
+    with wrap_new"""
+    if isinstance( item, np.ndarray):
+        return lambda shape: np.zeros( shape, dtype = item.dtype)
+    elif isinstance( item, RangesMatrix):
+        return RangesMatrix.zeros
+    elif isinstance( item, Ranges ):
+        def temp(shape):
+            assert len(shape) == 1
+            return Ranges( shape[0] )
+        return temp
+    elif isinstance( item, csr_array):
+        return lambda shape: csr_array(tuple(shape), dtype=item.dtype)
+    else:
+        raise ValueError(f"Cannot find zero type for {type(item)}")
+
+def _ranges_matrix_match( o, n, oidx, nidx):
+    """align RangesMatrix n entries to RangesMatrix o"""
+    assert len(oidx)==len(nidx)
+    if len(oidx) > 2:
+        raise NotImplemented
+    for i, x in zip( oidx[0], nidx[0]):
+        o.ranges[i] = _ranges_match( 
+            o.ranges[i], n.ranges[x],
+            [oidx[1]], [nidx[1]]
+        )
+    return o.copy()
+
+def _ranges_match( o, n, oidx, nidx):
+    """align Ranges n to Ranges o"""
+    assert len(oidx)==len(nidx)
+    assert len(oidx)==1
+    omsk = o.mask()
+    nmsk = n.mask()
+    omsk[oidx[0]] = nmsk[nidx[0]]
+    return Ranges.from_mask(omsk)
+
+def _expand(new, full, wrap_valid=True):
+    """new will become a top level axismanager in full once it is matched to
+    size"""
+    if 'dets' in new._axes:
+        _, fs_dets, ns_dets = full.dets.intersection(
+            new.dets, 
+            return_slices=True
+        )
+    if 'samps' in new._axes:
+        _, fs_samps, ns_samps = full.samps.intersection(
+            new.samps, 
+            return_slices=True
+        )
+    else:
+        fs_samps = slice(None)
+
+    out = full.copy(axes_only=True)
```

Comment:
were you going to make this a copy of only the axes that are in new? That should be pretty easy.

Reply:
Fixed this in [6fa2bd8](https://github.com/simonsobs/sotodlib/pull/751/commits/6fa2bd8da9bbfb04253883e03f4ae07884e501bf) and tested with updates. I think g2g now.

Reply:
added a suggestion b/c I don't think we need to explicitly pull out dets/samps?